From c7a9e55eb20cbaf12326bdffb57f881febe1e9c2 Mon Sep 17 00:00:00 2001
From: Jay Yang <jay.yang@intel.com>
Date: Fri, 2 Jul 2021 10:02:51 +0000
Subject: [PATCH] [VP] Fix 2pass CSC PROCAMP not work issue

Enable PROCAMP when 2pass CSC is also enabled.
Fixes: #1138
Signed off by: jay.yang@intel.com

This patch was importaed from iHD media-driver git server
(https://github.com/intel/media-driver.git) as of commit id
06905e60e8aacc86828e79b1431188ba6ed46268

Upstream-status: Backport
Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 media_driver/agnostic/common/vp/hal/vphal_render_vebox_base.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media_driver/agnostic/common/vp/hal/vphal_render_vebox_base.cpp b/media_driver/agnostic/common/vp/hal/vphal_render_vebox_base.cpp
index edea839e..20634286 100644
--- a/media_driver/agnostic/common/vp/hal/vphal_render_vebox_base.cpp
+++ b/media_driver/agnostic/common/vp/hal/vphal_render_vebox_base.cpp
@@ -2502,6 +2502,7 @@ void VPHAL_VEBOX_STATE::VeboxSetRenderingFlags(
                                         pSrc->ColorSpace != pRenderTarget->ColorSpace);
 
     pRenderData->bProcamp           = ((IS_VPHAL_OUTPUT_PIPE_VEBOX(pRenderData) ||
+                                        pRenderData->b2PassesCSC                ||
                                         IS_VPHAL_OUTPUT_PIPE_SFC(pRenderData))  &&
                                         pSrc->pProcampParams                    &&
                                         pSrc->pProcampParams->bEnabled);
-- 
2.17.1

